plugins/VstEffect/VstEffectControlDialog.cpp

```diff
@@ -81,9 +81,12 @@ VstEffectControlDialog::VstEffectControlDialog( VstEffectControls * _ctl ) :
 		if (embed_vst) {
 			btn->setCheckable( true );
 			btn->setChecked( true );
-		}
-		connect( btn, SIGNAL( toggled( bool ) ),
+			connect( btn, SIGNAL( toggled( bool ) ),
 					SLOT( togglePluginUI( bool ) ) );
+		} else {
+			connect( btn, SIGNAL( clicked() ),
+					m_plugin.data(), SLOT( toggleUI() ) );
```

Comment:
Why was this change needed? Can we fix it within `togglePluginUI` or `toggleUI` instead?

Reply:
> The `toggled(bool)` signal isn't emitted if the button isn't checkable

If we can uncheck the button when VST window is closed, we can always make the button checkable. AFAIK we do something similar for ZynAddSubFX.